textworld/envs/wrappers/filter.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+
+from typing import Tuple, Mapping, Any
+
+from textworld.core import GameState, Wrapper
+
+
+class Filter(Wrapper):
+    """
+    Environment wrapper to filter what information is made available.
+    """
+
+    def __init__(self, request_infos) -> None:
+        """
+        Arguments:
+            request_infos:
+                Specify which additional information from the `GameState` object
+                should be available in the `infos` dictionary returned by
+                `env.reset()` and `env.step()`. Possible choices are:
+
+                 * `'description'`: text description of the current room,
```

Comment:
An enum might be nice (https://docs.python.org/3/library/enum.html)

Reply:
(The docs say `extras:`; the code says `extra:`.)

You could do something like

```python
from enum import Enum

class GameInfo(Enum):
    DESCRIPTION = "description"
    INVENTORY = "inventory"
    MAX_SCORE = "max_score"
    # etc.
```

But if there's an unlimited number of extras you might want access to, then this won't work.

Reply:
Right, the extras are unlimited.
Thanks for the typo "extra**s**".

Reply:
Ideally, you'd need a way of asking for which information is available.

Reply:
Also, to give you an idea of what extra can be. In the games for the competition, agents can request `extra:recipe` (content of the cookbook) or `extra:walkthrough` (one possible sequence of commands that leads to winning the game).

Reply:
Yeah then this is probably as good as we can do.  Might be nice to avoid duplicating the docs a bunch though.

Reply:
Yes, I know. I find this annoying too and very difficult to maintain. :(

Reply:
`EnvironmentOptions`?  That sounds great to me.